### PR TITLE
repeatWithBehavior / retryWithBehavior

### PR DESCRIPTION
**Pull request by arnauddorgans**
#125

### CODE REVIEW

**Comment by freak4pc** [comment]
@Arnoymous Please add a description of the changes you made to the PR body and please give a proper commit message to your commit. Thanks! :)

File: Source/RxSwift/repeatWithBehavior.swift [hunk 1]
```diff
@@ -40,37 +35,31 @@ extension ObservableType {
...
-                    // otherwise retry after specified delay
-                    return Observable<Void>.just(()).delaySubscription(conditions.delay, scheduler: scheduler).flatMapLatest {
-                        self.repeatWithBehavior(currentRepeat + 1, behavior: behavior, scheduler: scheduler, shouldRepeat: shouldRepeat)
+                    var observable: Observable<E> = self.asObservable() // if there is no delay, simply retry
+                    if conditions.delay > 0.0 && currentRepeat > defaultRepeat {
+                        // otherwise retry after specified delay
+                        observable = observable.delaySubscription(conditions.delay, scheduler: scheduler)
                     }
+                    return observable.do(onCompleted: {
+                        DispatchQueue.main.async {
```
Comment (fpillet) [inline Source/RxSwift/repeatWithBehavior.swift]: You are not respecting the scheduler that was requested in the operator call. This should be scheduled on `scheduler`.

File: Source/RxSwift/retryWithBehavior.swift [hunk 2]
```diff
@@ -72,32 +72,38 @@ extension ObservableType {
...
+                        
+                        if conditions.delay > 0.0 && currentAttempt > defaultAttempt {
+                            // otherwise retry after specified delay
+                            observable = observable.delaySubscription(conditions.delay, scheduler: scheduler)
+                        }
+                    }
+                    return observable
+                        .map { ($0, nil) }
+                        .do(onError: { error in
+                            DispatchQueue.main.async {
```
Comment (fpillet) [inline Source/RxSwift/retryWithBehavior.swift]: You are not respecting the scheduler that was requested in the operator call. This should be scheduled on scheduler.

**Comment by Arnoymous** [comment]
well it's more than just a memory leak, add this on your app and wait for a crash
```swift
Observable.just(1)
            .repeatWithBehavior(.immediate(maxCount: .max))
            .debug()
            .subscribe()
            .disposed(by: disposeBag)
```
I've changed the signature and set the right scheduler but without an async scheduler we have warnings about cyclic dependency :(

**Comment by MortyMerr** [comment]
Close due to lack of activity and unclear PR purposes. The initial issue was closed more than a year ago.

**Pull request closed without merging**